go/vt/mysqlctl/mysqld.go

```diff
@@ -107,9 +112,92 @@ func NewMysqld(dbcfgs *dbconfigs.DBConfigs) *Mysqld {
 	result.appPool = dbconnpool.NewConnectionPool("AppConnPool", *appPoolSize, *appIdleTimeout, *poolDynamicHostnameResolution)
 	result.appPool.Open(dbcfgs.AppWithDB(), appMysqlStats)
 
+	version, getErr := getVersionString()
+	f, v, err := parseVersionString(version)
+
+	// Fallback if required
+	if getErr != nil || err != nil {
+		f, v, err = getVersionFromEnv()
+		if err != nil {
+			panic("Could not detect version from mysqld --version or MYSQL_FLAVOR")
+		}
+
+	}
+
+	log.Infof("Using flavor: %v, version: %v", f, v)
+	result.capabilities = NewCapabilitySet(f, v)
 	return result
 }
 
+func getVersionFromEnv() (flavor mysqlFlavor, ver serverVersion, err error) {
```

Comment:
Add a function doc comment explaining the caveats. Something like:

```go
/*
getVersionFromEnv returns the flavor and an assumed version based on the legacy
MYSQL_FLAVOR environment variable.

The assumed version may not be accurate since the legacy variable only specifies
broad families of compatible versions. However, the differences between those
versions should only matter if Vitess is managing the lifecycle of mysqld, in which
case we should have a local copy of the mysqld binary from which we can fetch
the accurate version instead of falling back to this function (see getVersionString).
*/
```